### ISSUE TITLE
Show caption in media details

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#2337]
from Structured Commons.
Show on first line, before title, in user's language or fallback language.

**Comment by nicolas-raoul** [comment]
@vanshikaarora see #2096

**Comment by vanshikaarora** [comment]
Hey @nicolas-raoul as mentioned by here https://github.com/commons-app/apps-android-commons/issues/2297#issuecomment-453988747. Shall I include the steps from to or should i just change the filename and make it multilingual?

**Comment by nicolas-raoul** [comment]
@vanshikaarora Can you see the "Captions" section at https://commons.wikimedia.org/wiki/File:R%C3%A9union_publique_de_La_R%C3%A9publique_En_Marche_!_Japon,_avec_le_d%C3%A9put%C3%A9_C%C3%A9dric_Villani_le_jeudi_26_novembre_de_%C3%A0_Akasaka,_Tokyo.jpg ?

The idea is to implement the same section in the app.
Search for this picture in the app, and you will see that captions are not shown.

**Comment by vanshikaarora** [comment]
>Rename the current filename field to "Caption"
Make it multilingual
When uploading, just take the first caption and use it as the filename

@nicolas-raoul So shall I start working upon it this way?

**Comment by vanshikaarora** [comment]
So, I can add the field for caption in the UI of that activity.
But from where will I get the caption for the image being viewed? 
Shall I add an additional field for caption while uploading the image itself?

**Comment by nicolas-raoul** [comment]
We don't know either, so you will have to look up Mediawiki documentation. You might have to search for a few hours, good luck!

**Comment by vanshikaarora** [comment]
OK I'll see the API's currently available

**Comment by vanshikaarora** [comment]
Hey @nicolas-raoul I have gone through the API's available on the documentation [here](https://commons.wikimedia.org/w/api.php). For our concern I would this Upload [API](https://commons.wikimedia.org/w/api.php?action=help&modules=upload). OUt of these we can use either if comment or text parameter to send the caption that will be added by user while uploading any image.

Please give me suggestions whether I am working on this correctly or not

**Comment by vanshikaarora** [comment]
Hi @iadeelzafar actually I had gone through the documentation and other resources for working on this issue. And was awaiting for suggestions. Well if you can also pitch in some suggestions then it would be great. Or if you already know how to work on this then you can go ahead with this issue:)

**Comment by brendajerop** [comment]
@nicolas-raoul I may have found a way to extract captions:
1. Get the contents of the page containing the file- the response is html
2. Extract the captions from the html string using [Jsoup](https://jsoup.org/)
Tried with a test app and it worked
![20190306_183753](https://user-images.githubusercontent.com/20903830/53894318-f1c7d280-4040-11e9-993e-bb343a95ef47.png)
![screenshot_2019-03-06-18-33-35_1](https://user-images.githubusercontent.com/20903830/53894317-f1c7d280-4040-11e9-9c6b-0013083a77f0.png)

**Comment by nicolas-raoul** [comment]
@brendajerop That's an option in case there is no API :-)

I am pretty sure there must be some kind of API, though... that would be ideal in terms of speed and reliability :-)

**Comment by brendajerop** [comment]
@nicolas-raoul You're right. There has to be a cleaner way to do this. I'll keep looking

**Comment by brendajerop** [comment]
@nicolas-raoul So I asked [here](https://discourse-mediawiki.wmflabs.org/t/how-to-get-captions-from-a-file/1207) and when I implemmented the solution, the response was...
![captions2](https://user-images.githubusercontent.com/20903830/53960437-60676780-40f7-11e9-8078-ab90cf662144.png)
... which if you make pretty becomes
![prettycaption](https://user-images.githubusercontent.com/20903830/53960465-74ab6480-40f7-11e9-95f3-ea28ac6deaf8.png)

**Comment by nicolas-raoul** [comment]
@brendajerop Wonderful! :-) It paste here the URL for reference:
https://commons.wikimedia.org/w/api.php?action=wbgetentities&sites=commonswiki&titles=File%3AR%C3%A9union_publique_de_La_R%C3%A9publique_En_Marche_!_Japon,_avec_le_d%C3%A9put%C3%A9_C%C3%A9dric_Villani_le_jeudi_26_novembre_de_%C3%A0_Akasaka,_Tokyo.jpg&props=labels&languages=fr&languagefallback=1

**Comment by noahwalugembe** [comment]
Hi @brendajerop iam also interested in working on this

**Comment by nicolas-raoul** [comment]
@brendajerop What is the status on this? :-)

**Comment by noahwalugembe** [comment]
Is this still available

**Comment by brendajerop** [comment]
I don't think this was ever assigned to me so it's open. I was only helping in regards to getting the captions:)

**Comment by nicolas-raoul** [comment]
@noahwalugembe I suggest you start working on #1107 and then get another issue after #1107 is done :-)

**Comment by sp2710** [comment]
@nicolas-raoul Shall I pick it up?

**Comment by namrata-jha** [comment]
Can I work on this issue please?

**Comment by nicolas-raoul** [comment]
@sp2710 It is yours, thanks!
